src/components/WalletInit/RestoreWallet/VerifyRestoredWallet.js

```diff
@@ -0,0 +1,156 @@
+    defaultMessage: '!!!Be careful about wallet restoration:',
+    description: 'some desc',
+  },
+  instructions1: {
+    id: 'components.walletinit.verifyrestoredwallet.instructionLabel-1',
+    defaultMessage: '!!!Wallet Instruction',
+    description: 'some desc',
+  },
+  instructions2: {
+    id: 'components.walletinit.verifyrestoredwallet.instructionLabel-2',
+    defaultMessage: '!!!Wallet Instruction',
+    description: 'some desc',
+  },
+  instructions3: {
+    id: 'components.walletinit.verifyrestoredwallet.instructionLabel-3',
+    defaultMessage: '!!!Wallet Instruction',
+    description: 'some desc',
+  },
+  walletAddressLabel: {
+    id: 'components.walletinit.verifyrestoredwallet.walletAddressLabel',
+    defaultMessage: '!!!Wallet Instruction',
+    description: 'some desc',
+  },
+  buttonText: {
+    id: 'components.walletinit.verifyrestoredwallet.buttonText',
+    defaultMessage: '!!!Wallet Instruction',
+    description: 'some desc',
+  },
+})
+
+const walletRestaurationIntruction = (formatMessage) => [
+  formatMessage(messages.instructions1),
+  formatMessage(messages.instructions2),
+  formatMessage(messages.instructions3),
+]
+
+const walletAddresses = [
+  'DdfFfGgXxCcXcV … hHhJKKHgfdaa',
+  'AaAaSdDD4534 … kdDDdDdfFfGgXx',
+]
+
+const RestoreWalletScreen = ({
+  navigateToWalletCredentials,
+  intl,
+  navigation,
+  checksum,
+}) => {
+  const {formatMessage} = intl
+  return (
+    <SafeAreaView style={styles.safeAreaView}>
+      <StatusBar type="dark" />
+      <View>
+        <Text style={styles.textStyles}>
+          {formatMessage(messages.checksumLabel)}
+        </Text>
+        <CheckSumView avatar={mockAvatar} checksum={checksum} />
+        <Text style={styles.titleStyles}>
+          {formatMessage(messages.instructionLabel)}
+        </Text>
+        <FlatList
+          data={walletRestaurationIntruction(formatMessage)}
+          keyExtractor={(item) => item}
+          renderItem={({item}) => (
+            <BulletPointItem textRow={item} style={styles.instructionStyles} />
+          )}
+        />
+        <View style={styles.addressesStyles}>
+          <Text style={styles.titleStyles}>
+            {formatMessage(messages.walletAddressLabel)}
+          </Text>
+          <FlatList
+            data={walletAddresses}
+            keyExtractor={(item) => item}
+            renderItem={({item}) => <WalletAddress addressHash={item} />}
+          />
+        </View>
+      </View>
+      <Button
+        onPress={navigateToWalletCredentials}
+        title={formatMessage(messages.buttonText)}
+      />
+    </SafeAreaView>
+  )
+}
+
+export default injectIntl(
+  (compose(
+    connect((state) => ({})),
+    withNavigation,
+    withNavigationTitle(({intl}) => intl.formatMessage(messages.title)),
+    withStateHandlers(
+      {
+        checksum: 'BONE-0770',
+        walletNumber: '123455', // added wallet number
+        // not sure about the phrase from restore wallet screen
```

Comment:
not sure what this comment means. However, it probably shouldn't be hard-coded here.

Reply:
Hi 😃 , I was not sure about this - the current screen (VerifyRestoreVallet) is placed between Restored wallet (the one in which user enter a 15-word recovery phrase) and the Wallet credentials (where user inputs wallet name and the passwoard ) . I'm not sure about how the integration with the backend will be and  if the wallet credential screen should have the recovery phrase or not (if I should pass it to it).

Reply:
https://github.com/cardano-foundation/CIPs/blob/21d1a289efea4b8bef179ca75d0e61ef377bf22c/CIP4/CIP4.md this document describes exactly how it works and it has a reference implementation linked at the bottom. You can see how we use CIP4 in the Yoroi extension to see how we generate the image from the JS library.